### PR TITLE
FC-592 add accordion component

### PR DESCRIPTION
**Pull request by zoheiry**
## Purpose
Create Accordion component

### Visualization
screen shot 2018-10-04 at 5 21 09 pm

### Links
- [Jira Ticket](https://fashiontrade.atlassian.net/browse/FC-592)

### Checklist
- [x] Discuss best solution for using Accordion container

### CODE REVIEW

File: src/components/Accordion/Accordion.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    this.setState({ open: newOpenState });
+  }
+
+  modifyChildren = () => {
+    const { children } = this.props;
+    if (children && children.length > 1) {
+      let itemIndex = -1;
+      return children.map(child => {
+        const isAccordionItem = get(child, 'type.displayName') === 'AccordionItem';
+        if (isAccordionItem) {
```
Comment (frontalicious-martijn) [inline src/components/Accordion/Accordion.js]: Why do we want to allow other types of items in this component? Each item could contain whatever content right?

File: src/components/Accordion/Accordion.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+class Accordion extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      open: props.initialOpen || {}
+    };
+  }
+
+  handleItemClick = (defaultClickHandler, itemIndex) => {
```
Comment (frontalicious-martijn) [inline src/components/Accordion/Accordion.js]: Maybe it is possible to move the open/close state into the item? These items could report back to the Accordion if it is clicked. That way the accordion could a close event to the other items? Only the ones that are open will react to that?

File: src/components/AccordionItem/AccordionItem.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+AccordionItem.displayName = 'AccordionItem';
+
+AccordionItem.defaultProps = {
+  onClick: () => null
+};
+
+AccordionItem.propTypes = {
+  isDisabled: PropTypes.bool,
```
Comment (frontalicious-martijn) [inline src/components/AccordionItem/AccordionItem.js]: I wonder is we should get rid of the `is` prefix for boolean values. Just like normal html (disable, required, selected). Same for isOpen. What do you think?

File: src/components/index.js [hunk 4]
```diff
@@ -9,3 +9,5 @@ export LoadingSpinner from './LoadingSpinner';
 export Searchbar from './Searchbar';
 export Modal from './Modal';
 export CheckBox from './CheckBox';
+export Accordion from './Accordion';
+export AccordionItem from './AccordionItem';
```
Comment (frontalicious-martijn) [inline src/components/index.js]: Do we want to expose this here? Or only Accordion.Item?

File: src/components/AccordionItem/AccordionItem.js (hunk 3 above)
Comment (kirill-kruchkov) [reply src/components/AccordionItem/AccordionItem.js]: I would love it that way too

**Pull request merged**